client/dashboard/leaderboards/index.js

```diff
@@ -0,0 +1,184 @@
+
+/**
+ * Internal dependencies
+ */
+import withSelect from 'wc-api/with-select';
+import TopSellingProducts from './top-selling-products';
+import './style.scss';
+
+class Leaderboards extends Component {
+	constructor( props ) {
+		super( ...arguments );
+		this.state = {
+			hiddenLeaderboardKeys: props.userPrefLeaderboards || [],
+			rowsPerTable: props.userPrefLeaderboardRows || 5,
+		};
+
+		this.toggle = this.toggle.bind( this );
+	}
+
+	componentDidUpdate( {
+		userPrefLeaderboardRows: prevUserPrefLeaderboardRows,
+		userPrefLeaderboards: prevUserPrefLeaderboards,
+	} ) {
+		const { userPrefLeaderboardRows, userPrefLeaderboards } = this.props;
+		if ( userPrefLeaderboards && ! isEqual( userPrefLeaderboards, prevUserPrefLeaderboards ) ) {
+			/* eslint-disable react/no-did-update-set-state */
+			this.setState( {
+				hiddenLeaderboardKeys: userPrefLeaderboards,
+			} );
+			/* eslint-enable react/no-did-update-set-state */
+		}
+		if (
+			userPrefLeaderboardRows &&
+			parseInt( userPrefLeaderboardRows ) !== parseInt( prevUserPrefLeaderboardRows )
+		) {
+			/* eslint-disable react/no-did-update-set-state */
+			this.setState( {
+				rowsPerTable: parseInt( userPrefLeaderboardRows ),
+			} );
+			/* eslint-enable react/no-did-update-set-state */
+		}
+	}
+
+	toggle( key ) {
+		return () => {
+			this.setState(
+				prevState => {
+					const hidden = prevState.hiddenLeaderboardKeys.includes( key );
+					const hiddenLeaderboardKeys = prevState.hiddenLeaderboardKeys.filter(
+						chartKey => chartKey !== key
+					);
+
+					if ( ! hidden ) {
+						hiddenLeaderboardKeys.push( key );
+					}
+					return { hiddenLeaderboardKeys };
+				},
+				() => {
+					const userDataFields = {
+						[ 'dashboard_leaderboards' ]: this.state.hiddenLeaderboardKeys,
+					};
+					this.props.updateCurrentUserData( userDataFields );
+				}
+			);
+		};
+	}
+
+	setRowsPerTable = rows => {
+		return this.setState(
+			{
+				rowsPerTable: parseInt( rows ),
+			},
+			() => {
+				const userDataFields = {
+					[ 'dashboard_leaderboard_rows' ]: this.state.rowsPerTable,
+				};
+				this.props.updateCurrentUserData( userDataFields );
+			}
+		);
+	};
+
+	renderMenu() {
+		const { hiddenLeaderboardKeys, rowsPerTable } = this.state;
+		const allLeaderboards = [
+			{
+				key: 'top-products',
+				label: __( 'Top Products', 'wc-admin' ),
+			},
+			{
+				key: 'top-categories',
+				label: __( 'Top Categories', 'wc-admin' ),
+			},
+			{
+				key: 'top-coupons',
+				label: __( 'Top Coupons', 'wc-admin' ),
+			},
+		];
+		return (
+			<EllipsisMenu label={ __( 'Choose which leaderboards to display', 'wc-admin' ) }>
+				<Fragment>
+					<MenuTitle>{ __( 'Leaderboards', 'wc-admin' ) }</MenuTitle>
+					{ allLeaderboards.map( leaderboard => {
+						return (
+							<MenuItem onInvoke={ this.toggle( leaderboard.key ) } key={ leaderboard.key }>
+								<ToggleControl
+									label={ __( `${ leaderboard.label }`, 'wc-admin' ) }
```

Comment:
The leaderboard labels are already passed through i18n.

Reply:
You're right, good catch!  Updated.